2022 Jamstack Chapter (#3214)
HTTPArchive/almanac.httparchive.org: src/content/en/2022/jamstack.md

---
#See https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters
title: Jamstack
description: Jamstack chapter of the 2022 Web Almanac covers quantifying Jamstack sites, the growth of Jamstack, Jamstack-y frameworks and hosting.
authors: [seldo, whitep4nth3r]

> reviewers: [GregBrimble, tunetheweb]

Reviewer: ```suggestion
reviewers: [tunetheweb]
```

#TODO - Review and update chapter description
description: Jamstack chapter of the 2022 Web Almanac covers adoption of technology, performance of websites built with Jamstack and weights for various resources.
authors: [seldo, whitep4nth3r]
reviewers: [GregBrimble, Nutlope, tunetheweb]
analysts: [seldo, tunetheweb]

> editors: [DesignrKnight]

Author: @seldo @whitep4nth3r could you confirm these are the contributors, listed in descending order of contribution? and that we haven't missed anyone?

Reviewer 2: No one is missing, to my knowledge!

Author: Did @GregBrimble and @Nutlope contribute? Can't remember now but, if not, then we should remove.

Reviewer 3: @tunetheweb I did not contribute actually, feel free to remove me.

Author: Done. Thanks for letting us know. Maybe next year!

Reviewer: Likewise, I couldn't this year either unfortunately.

translators: []

> seldo_bio: Laurie has been a web developer since 1996, with occasional breaks to found companies like <a hreflang="en" href="https://www.crunchbase.com/organization/snowball-factory">awe.sm</a> (2010) and <a hreflanh="en" href="https://seldo.com/about/#:~:text=been%20a%20web,learn%20so%20much.">npm</a> (2014). He likes making websites, and he likes making the web bigger and better. He thinks one of the best ways to do that is to encourage more people to do web development, by teaching them existing techniques and by building tools and services that make web development easier, so they don't have to learn so much.

Author: @seldo, I pulled this from your website, but feel free to change as you see fit. This is listed at the bottom of the chapter.

whitep4nth3r_bio: Salma helps developers build stuff, learn things and love what they do. She works at Netlify as a Staff Developer Experience Engineer, streams live coding, and loves helping people get into tech. After a career as a music teacher and comedian, Salma transitioned to technology in 2014, specializing in front end development and tech leadership for startups, agencies and global e-commerce. <a hreflang="en" href="https://twitch.tv/whitep4nth3r">Find Salma on Twitch</a> to see what she's currently building.
results: https://docs.google.com/spreadsheets/d/1yfNaj25ToezMwQLKdYP6Qh7AUoX9zMdKMSRVC8JlZMY/
featured_quote: TODO

> featured_stat_1: TODO

Author: @seldo @whitep4nth3r we need one Featured Quote, and 3 stats to highlight on the home page (at random, along with the other chapters).

Here's last year's ones:

```
featured_quote: Jamstack young and fast growing technology to build modern websites. That what people say and that's proved by numbers.
featured_stat_1: 1.1%
featured_stat_label_1: Jamstack websites adoption among web
featured_stat_2: 9.5%
featured_stat_label_2: Median value of jQuery usage among Jamstack sites
featured_stat_3: 1.7MB
featured_stat_label_3: Median value of resource wights among Jamstack sites
```

Which looked [like this on the home page](https://almanac.httparchive.org/en/2021/?feat=jamstack#featured-chapter):

<img width="655" alt="image" src="https://user-images.githubusercontent.com/10931297/195334113-f9012268-163a-4bc4-b826-4903825e2e94.png">

Reviewer 2: Some suggestions for the featured quote:

> Although we can’t claim to know exactly what percentage of the web is Jamstack, we can say that around 3% of the web is Jamstack-y, and that this group has been growing strongly for the last 3 years, a great sign for the Jamstack community.

> We believe you can achieve Jamstack-y results with any tech stack, and we hope you find this data useful in identifying tools and platforms that might make it easier to achieve.

Some suggestions for the stats:

```
featured_stat_1: ~3%
featured_stat_label_1: Jamstack-y sites on the web in 2022
featured_stat_2: >100%
featured_stat_label_2: growth of Jamstack-y sites since 2020
featured_stat_3: 36%
featured_stat_label_3: of Jekyll sites are Jamstack-y  
```

Author: SGTM!

Have added the first quote and those stats (with slight tweaks) in https://github.com/HTTPArchive/almanac.httparchive.org/pull/3214/commits/1dc3650d3ab0bd8bcab7f2a9f5cdaaec3b5dd963

It looks like this:

<img width="632" alt="image" src="https://user-images.githubusercontent.com/10931297/195342455-9f7bd652-0164-49f1-9fae-f28183b40121.png">

Reviewer 2: That's great, thanks @tunetheweb!

Author: Actually changed the 100% to 106% (which is the mobile number) as feels more "real" when not rounded:

<img width="644" alt="image" src="https://user-images.githubusercontent.com/10931297/195343839-cd9a5f33-48e9-460f-9329-4c45f046a0e6.png">

Reviewer 2: Great!

But as the emphasized words demonstrate, there's clearly some continuity: the sites should be fast, they should be pre-rendered, and they should use an architectural approach that decouples "where you get your data" from "how you render your data". Even if a precise dictionary definition is hard to come by, Jamstack developers know what you mean when you say "Jamstack": you've got a site that loads really quickly, renders a lot of its useful content once, at build time, and retrieves additional data (if it needs to) via JavaScript.

<p class="note">Disclosure: the two authors of this report were Netlify employees. This report and the underlying analysis were reviewed and approved by multiple volunteers not affiliated with Netlify.</p>

## Quantifying the Jamstack: what counts?

But the problem gets more tricky when you're trying to put together the 2022 Web Almanac. When you're dealing with millions of websites, "I know it when I see it" can't be your definition. How do we quantify the Jamstack? How do we precisely identify it so we can learn about it? We started by asking ourselves a series of questions.

### Is every static site a Jamstack site?

> That seems like it should be an obvious "yes": if the page is flat HTML that renders all at once then it certainly sounds like it should be Jamstack. But what about all those pages people built in the 90s, before JavaScript was popular and before servers did much? Are they Jamstack? It felt like they weren't, not every static site is a Jamstack site. So we tried to think of why.

Author: I'm not sure that's what @seldo meant by this? I think he meant Web 1.0 websites that were static, before dynamic, interactive sites came along when servers moved from serving static sites to more dynamic content?

> That seems like it should be an obvious "yes": if the page is flat HTML that renders all at once then it certainly sounds like it should be Jamstack. But what about all those pages people built in the 90s, before JavaScript was popular and before servers did much? Are they Jamstack? It felt like they weren't, not every static site is a Jamstack site. So we tried to think of why.

Reviewer 2: Yes that's exactly it.

> That seems like it should be an obvious "yes": if the page is flat HTML that renders all at once then it certainly sounds like it should be Jamstack. But what about all those pages people built in the 90s, before JavaScript was popular and before servers did much? Are they Jamstack? It felt like they weren't, not every static site is a Jamstack site. So we tried to think of why.

Author: What about this:

```suggestion
That seems like it should be an obvious "yes": if the page is flat HTML that renders all at once then it certainly sounds like it should be Jamstack. But what about all those pages people built in the 90s, before JavaScript was popular and before sites were as dynamic as they are now? Are they Jamstack? It felt like they weren't, not every static site is a Jamstack site. So we tried to think of why.
```

> That seems like it should be an obvious "yes": if the page is flat HTML that renders all at once then it certainly sounds like it should be Jamstack. But what about all those pages people built in the 90s, before JavaScript was popular and before servers did much? Are they Jamstack? It felt like they weren't, not every static site is a Jamstack site. So we tried to think of why.

Reviewer 2: Hmm, actually this might complicate things given the large portion of *static sites* the report covers.

How about mentioning the option to choose between whether you're rendering static files or serving dynamic (i.e. personalised) content?

> That seems like it should be an obvious "yes": if the page is flat HTML that renders all at once then it certainly sounds like it should be Jamstack. But what about all those pages people built in the 90s, before JavaScript was popular and before servers did much? Are they Jamstack? It felt like they weren't, not every static site is a Jamstack site. So we tried to think of why.

Reviewer 2: The main point I think is about the options we have. Monoliths. Serverless. Edge. Static. Hybrid. All the other stuff in between.

> That seems like it should be an obvious "yes": if the page is flat HTML that renders all at once then it certainly sounds like it should be Jamstack. But what about all those pages people built in the 90s, before JavaScript was popular and before servers did much? Are they Jamstack? It felt like they weren't, not every static site is a Jamstack site. So we tried to think of why.

Author: Maybe just leave as is then? I think it's understandable.
